lightning_ir/retrieve/plaid/plaid_searcher.py

```diff
@@ -167,28 +54,29 @@ def search(self, output: BiEncoderOutput) -> Tuple[PackedTensor, List[List[str]]
             Tuple[PackedTensor, List[List[str]]]: A tuple containing the scores and the corresponding document IDs.
         Raises:
             ValueError: If the output does not contain query embeddings.
+            ValueError: If the index is not loaded. Call load() before searching.
         """
-        query_embeddings = output.query_embeddings
-        if query_embeddings is None:
+        if output.query_embeddings is None:
             raise ValueError("Expected query_embeddings in BiEncoderOutput")
-        query_embeddings = query_embeddings.to(self.device)
 
-        _, candidate_idcs = self._centroid_candidate_retrieval(query_embeddings)
-        num_docs = candidate_idcs.lengths
+        if not self.index:
+            raise ValueError("Index not loaded. Call load() before searching.")
 
-        # compute scores
-        doc_embeddings = self._reconstruct_doc_embeddings(candidate_idcs)
-        output.doc_embeddings = doc_embeddings
-        output = self.module.model.score(output, num_docs)
-        scores = output.scores
-        if scores is None:
-            raise ValueError("Expected scores in BiEncoderOutput")
+        scores = self.index.search(
+            queries_embeddings=output.query_embeddings.embeddings.detach(),
```

Comment:
Why the detach? The embeddings should, in theory, never have gradients at this point